services/src/datasets/listing.rs

```diff
@@ -0,0 +1,120 @@
+use crate::datasets::storage::{AddDataSetProvider, AddMockDataSetProvider, DataSet};
+use crate::error::Result;
+use crate::users::user::UserId;
+use crate::util::user_input::{UserInput, Validated};
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::DataSetId;
+use geoengine_operators::engine::{
+    MetaData, MetaDataProvider, ResultDescriptor, VectorResultDescriptor,
+};
+use geoengine_operators::mock::MockDataSetDataSourceLoadingInfo;
+use geoengine_operators::source::OgrSourceDataset;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct DataSetListing {
+    pub id: DataSetId,
+    pub name: String,
+    pub description: String,
+    pub tags: Vec<String>,
+    pub source_operator: String, // TODO: enum?
+                                 // TODO: meta data like bounds, crs, resolution
```

Comment:
I guess at least CRS is important since Johannes just recently enforced it.

Reply:
74bb6bae607bd98f778a77ea862b32fc132ae3f3